### PR TITLE
PWN-8111 - Fix pull to refresh state on home screen

### PR DESCRIPTION
**Pull request by Davran331122**
https://p2pvalidator.atlassian.net/browse/PWN-8111
https://p2pvalidator.atlassian.net/browse/PWN-8112

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/bridge/interactor/EthereumInteractor.kt [hunk 1]
```diff
@@ -10,10 +11,12 @@ import org.p2p.ethereumkit.external.repository.EthereumRepository
 import org.p2p.ethereumkit.internal.models.Signature
 import org.p2p.wallet.bridge.claim.interactor.ClaimInteractor
 import org.p2p.wallet.bridge.model.BridgeBundle
+import org.p2p.wallet.common.feature_toggles.toggles.remote.EthAddressEnabledFeatureToggle
 
 class EthereumInteractor(
     private val claimInteractor: ClaimInteractor,
     private val ethereumRepository: EthereumRepository,
+    private val ethAddressEnabledFeatureToggle: EthAddressEnabledFeatureToggle
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/interactor/EthereumInteractor.kt:19]: do you need this one?

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 2]
```diff
@@ -111,8 +111,11 @@ class HomePresenter(
                 view?.showRefreshing(true)
                 val solTokens = userInteractor.loadUserTokensAndUpdateLocal(tokenKeyProvider.publicKey.toPublicKey())
                 async { userInteractor.loadUserRates(solTokens) }
+                async {
+                    val bundles = ethereumInteractor.getListOfEthereumBundleStatuses()
+                    ethereumInteractor.loadWalletTokens(bundles)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:116]: how about making a method loadWalletTokens and make getListOfEthereumBundleStatuses call inside?
it looks like business logic to keep this methods together

File: app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt [hunk 3]
```diff
@@ -110,12 +107,20 @@ class UserTokensPolling(
     private suspend fun fetchSolTokens(): List<Token.Active> =
         userInteractor.loadUserTokensAndUpdateLocal(tokenKeyProvider.publicKey.toPublicKey())
 
-    private suspend fun fetchEthereumTokens(): List<Token.Eth> {
-        if (!ethAddressEnabledFeatureToggle.isFeatureEnabled) {
-            return emptyList()
+    private suspend fun fetchEthereumTokens() {
+        if (ethAddressEnabledFeatureToggle.isFeatureEnabled) {
+
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt:112]:
```suggestion
```

File: app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt [hunk 4]
```diff
@@ -110,12 +107,20 @@ class UserTokensPolling(
...
+            val ethBundles = ethereumInteractor.getListOfEthereumBundleStatuses()
+            ethereumInteractor.loadWalletTokens(ethBundles)
+        }
+    }
+
+    private fun getEthereumTokensFlow(): Flow<List<Token.Eth>> {
+        return if (ethAddressEnabledFeatureToggle.isFeatureEnabled) {
+            return ethereumInteractor.getTokensFlow()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt:120]: no need for return

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt:120]: maybe you can extract using ethAddressEnabledFeatureToggle inside interactor to remove duplicates?

File: ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumKitRepository.kt [hunk 5]
```diff
@@ -35,10 +37,12 @@ internal class EthereumKitRepository(
     private val tokensRepository: EthereumTokensRepository,
     private val priceRepository: PriceRepository,
     private val dispatchers: CoroutineDispatchers,
+
```
Comment (gslevinkov) [inline ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumKitRepository.kt:40]:
```suggestion
```

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt (hunk 2 above)
Comment (Davran331122) [reply app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:116]: okay

**Review by 1sakov — APPROVED** [review]

**Pull request merged**